esp/smc/SMCLib/TpContainer.cpp

```diff
@@ -743,11 +743,36 @@ static void refreshValidTargets()
     }
 }
 
+static void addUniqueDataPlaneName(const char * dataPlaneName)
+{
+    if (isEmptyString(dataPlaneName))
```

Comment:
I think if you are only iterating the storage.planes this will always be non null and unique.

Reply:
@ghalliday please see my comment https://github.com/hpcc-systems/HPCC-Platform/pull/14995#discussion_r648315662.